### PR TITLE
Change 'Test' so it immediately outputs failed test results

### PR DESCRIPTION
**Pull request by ChrisJefferson**
Decided to quickly look to see how hard it would be to implement #5209 . Turns out not too hard!

This should probably have some more tests (which would be added to testspecial I think, like the single existing one).

### CODE REVIEW

File: tst/testspecial/test-1.tst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+gap> a:=SymmetricGroup(5);;
+gap> # ... do some tests
+gap> a:=SymmetricGroup(1 0);;
+gap> Length(Elements(a)) = Factorial(10);
+true
```
Comment (fingolfin) [inline tst/testspecial/test-1.tst:5]:
```suggestion
true

```

File: tst/testspecial/testing-test.g.out [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+########> Diff in test-1.tst:4
+# Input is:
+Length(Elements(a)) = Factorial(10);
+# Expected output:
+true
+# But found:
+false
+########
+false
+gap> QUIT;
```
Comment (fingolfin) [inline tst/testspecial/testing-test.g.out:20]: So how does the output of this test differ from what one gets without your `Test` patch?

Comment (ChrisJefferson) [reply tst/testspecial/testing-test.g.out:20]: My hope is the output is identical, it just comes out faster

**Review by fingolfin — APPROVED** [review]
Works great for me!

**Pull request merged**